core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/PetSASpec.scala

```diff
@@ -0,0 +1,58 @@
+package org.broadinstitute.dsde.rawls.webservice
+
+import org.broadinstitute.dsde.rawls.dataaccess._
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{MockSamDAO, TestData}
+import org.broadinstitute.dsde.rawls.google.MockGooglePubSubDAO
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.rawls.openam.StandardUserInfoDirectives
+import spray.http._
+
+import scala.concurrent.ExecutionContext
+
+class PetSASpec extends ApiServiceSpec {
+  case class TestApiService(dataSource: SlickDataSource, user: RawlsUser, gcsDAO: MockGoogleServicesDAO, gpsDAO: MockGooglePubSubDAO, httpSamDAO: SamDAO)(implicit val executionContext: ExecutionContext) extends ApiServices with StandardUserInfoDirectives
+
+  def withApiServices[T](dataSource: SlickDataSource, user: RawlsUser = RawlsUser(userInfo))(testCode: TestApiService => T): T = {
+    val apiService = new TestApiService(dataSource, user, new MockGoogleServicesDAO("test"), new MockGooglePubSubDAO, new MockSamDAO)
+    try {
+      testCode(apiService)
+    } finally {
+      apiService.cleanupSupervisor
+    }
+  }
+
+  def withTestDataApiServices[T](testCode: TestApiService => T): T = {
+    withDefaultTestDatabase { dataSource: SlickDataSource =>
+      withApiServices(dataSource)(testCode)
+    }
+  }
+
+  it should "switch to User Account when Accessed from pet SA" in withMinimalTestDatabase { dataSource: SlickDataSource =>
+    withApiServices(dataSource) { services =>
+      val petSA = UserInfo(RawlsUserEmail("pet-123456789876543212345@gserviceaccount.com"), OAuth2BearerToken("token"), 123, RawlsUserSubjectId("123456789876"))
```

Comment:
@gserviceaccount.com isn't the right shape and the regex should barf here, no?

Reply:
Yeah.. it seems like the regex would not match here and the switch would not happen. So this test is probably _creating_ the petSA as a real user? I wonder if there is a different API we can use to better test this. Maybe:

1. Create a workspace as a user
2. Get the workspace as the pet
3. Verify the workspace in #1 is returned